server/web/controllers/auth.ex

```diff
@@ -51,3 +40,43 @@ defmodule Roommates.Auth do
     delete_session(conn, :user_id)
   end
 end
+
+defmodule Roommates.Auth.RequireLogin do
+  import Plug.Conn
+  import Phoenix.Controller
+
+  def init(opts) do
+    opts
+  end
+
+  def call(conn, _opts) do
+    if conn.assigns.current_user do
+      conn
+    else
+      conn
+      |> put_flash(:error, "You must be logged in to access that page")
+      |> redirect(to: "/")
+      |> halt()
+    end
+  end
+end
+
+defmodule Roommates.Auth.RequireGuest do
+  import Plug.Conn
+  import Phoenix.Controller
+
+  def init(opts) do
+    opts
+  end
+
+  def call(conn, _opts) do
+    if !conn.assigns.current_user do
```

Comment:
Sorry, I meant guard. It looks much better (and is more idiomatic to write this in this form:

```elixir
def call(%Plug.Conn{assigns: %{current_user: user}} = conn, _opts) when user, do: conn
def call(conn, _opts) do
    conn
    |> put_flash(:error, "You must be logged in to access that page")
    |> redirect(to: "/")
    |> halt()
  end
end
```